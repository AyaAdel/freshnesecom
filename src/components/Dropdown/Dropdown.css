.dropdown-list {
  position: relative;
  transition: all 0.3s ease-in-out;
}

.dropdown-list__items {
  max-height: 170px;
  overflow: auto;
  display: flex;
  flex-direction: column;
  border: 1px solid #d1d1d1;
  padding: 10px;
  border-radius: 5px;
  position: absolute;
  top: 45px;
  left: 0;
  font-size: 13px;
  background-color: #fff;
  z-index: 1;
  width: 150px;
}

.dropdown-list__items span {
  margin-bottom: 5px;
}

.dropdown-list__items span:hover {
  color: #6a983c;
  cursor: pointer;
}

.dropdown-list__list__selected {
  display: flex;
  cursor: pointer;
  justify-content: space-between;
  font-weight: 500;
  font-size: 15px;
  line-height: 22px;
}

.dropdown-list__list__selected img {
  margin-left: 8px;
  width: 12px;
}
