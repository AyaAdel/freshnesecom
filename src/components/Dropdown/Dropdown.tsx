import { useState } from 'react';
import ArrowDown from 'assets/chevron-down.svg';

import './Dropdown.css';

type DropdownProps = {
  defaultSelected: string;
  list: string[];
  onSelect?: (selected: string) => void;
};

export default function Dropdown({
  defaultSelected,
  list,
  onSelect,
}: DropdownProps) {
  const [isVisible, setIsVisible] = useState(false);
  const [selected, setSelected] = useState(defaultSelected);

  const toggleList = () => setIsVisible(!isVisible);

  const changeSelected = (selectedItem: string) => {
    setSelected(selectedItem);
    setIsVisible(false);
    onSelect && onSelect(selectedItem);
  };

  return (
    <div className="dropdown-list" data-testid="dropdown-list">
      <div className="dropdown-list__list__selected" onClick={toggleList}>
        <span data-testid="dropdown-list-selected">{selected}</span>
        <img src={ArrowDown} />
      </div>
      {isVisible && (
        <div className="dropdown-list__items">
          {list &&
            list.map((item: string, index: number) => (
              <span key={index} onClick={() => changeSelected(item)}>
                {item}
              </span>
            ))}
        </div>
      )}
    </div>
  );
}
