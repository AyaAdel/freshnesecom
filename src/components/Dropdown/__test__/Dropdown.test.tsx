import { render, screen, waitFor, act, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { RenderComponent } from 'utils/test/render';
import Dropdown from '../Dropdown';
import userEvent from '@testing-library/user-event';

function MyComponent() {
  return (
    <RenderComponent>
      <Dropdown defaultSelected="Title" list={['dropdown 1']} />
    </RenderComponent>
  );
}

describe('Dropdown', () => {
  afterEach(cleanup);

  it('Render Dropdown Correctly', () => {
    const tree = renderer.create(<MyComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Show Dropdown list onClick', async () => {
    act(() => {
      render(<MyComponent />);
    });

    userEvent.click(screen.getByTestId('dropdown-list-selected'));

    await waitFor(() => {
      expect(screen.getByText('dropdown 1')).toBeInTheDocument();
    });
  });

  it('Change Default Selected Title', async () => {
    act(() => {
      render(<MyComponent />);
    });

    userEvent.click(screen.getByTestId('dropdown-list-selected'));

    await waitFor(() => {
      screen.getByText('dropdown 1');
    });

    userEvent.click(screen.getByText('dropdown 1'));

    await waitFor(() => {
      expect(screen.getByTestId('dropdown-list-selected')).toHaveTextContent(
        'dropdown 1'
      );
    });
  });
});
