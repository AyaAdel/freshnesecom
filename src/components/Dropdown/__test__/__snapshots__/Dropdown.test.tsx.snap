// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Render Dropdown Correctly 1`] = `
<div
  className="dropdown-list"
  data-testid="dropdown-list"
>
  <div
    className="dropdown-list__list__selected"
    onClick={[Function]}
  >
    <span
      data-testid="dropdown-list-selected"
    >
      Title
    </span>
    <img
      src="chevron-down.svg"
    />
  </div>
</div>
`;
